connect/mirror/src/main/java/org/apache/kafka/connect/mirror/ConfigPropertyFilter.java

```diff
@@ -25,8 +25,20 @@
 @InterfaceStability.Evolving
 public interface ConfigPropertyFilter extends Configurable, AutoCloseable {
 
+    /**
+     * Specifies whether to replicate the given topic configuration.
+     * This does not handle replication of configurations with default values.
+     * The configurations with default values are controlled by {@link #shouldReplicateSourceDefault(String)}.
+     */
     boolean shouldReplicateConfigProperty(String prop);
 
+    /**
+     * Specifies whether to replicate the given topic configuration that has a default value
```

Comment:
I think we can give more detail on the exact behavior for this API:

```suggestion
     * Specifies how to replicate the given topic configuration property
     * that has a default value on the source cluster. Only invoked for properties
     * that {@link #shouldReplicateConfigProperty(String)} has returned
     * {@code true} for.
     *
     * @return {@code true} if the default value from the source topic should be synced
     * to the target topic, and {@code false} if the default value for the target topic
     * should be used instead
```